docs/reference/filebeat/file-identity.md

```diff
@@ -0,0 +1,183 @@
+
+It's not only `inode` values: when using the Linux [LVM](https://en.wikipedia.org/wiki/Logical_Volume_Manager_%28Linux%29) (Logical Volume Manager), device numbers are allocated dynamically at module load (refer to [Persistent Device Numbers](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/logical_volume_manager_administration/lv#persistent_numbers) in the Red Hat Enterprise Linux documentation).
+
+However, this problem with device numbers can be fixed by our next file identity option.
+
+### `inode_marker`
+
+Let's say you verified that your file system does not re-use `inode` values and it's safe to rely on them. But you found out that the device number was changing and caused Filebeat to re-ingest all the data from the beginning causing data duplication.
+
+This can be fixed by using a special marker file on your file system instead of asking the OS for a device number.
+
+Users specify a path by setting `file_identity.inode_marker.path`. This path leads to a file that has some short unique text that remains unchanged forever. The content of this file replaces the `device` number in the `native` file identity. The rest remains the same as `native`.
+
+### `fingerprint`
+
+The most stable file identity implementation at the moment. This file identity is default since Filebeat version 9.0.
+
+This file identity is using file fingerprints produced by the scanner component of the filestream input. Therefore, in order to activate this file identity, it's required to first activate fingerprint in the scanner by setting `prospector.scanner.fingerprint.enabled: true`.
+
+In the context of the filestream input, fingerprint is a hash string computed from a part of the file content.
+
+![Fingerprint File Identity](/reference/filebeat/img/fingerprint-file-identity.png)
+
+By default, the hashed part of the file is the first 1024 bytes (`offset: 0`, `length: 1024`). Most of the log files have timestamps and other unique segments on each line that make sure that the first 1024 bytes combined together constitute a unique set of characters. This makes sure that a produced fingerprint is a stable unique identifier for each file.
+
+If necessary it's possible to move this hashed byte range farther in the file by adjusting `prospector.scanner.fingerprint.offset`. It's also possible to change the amount of bytes used for hashing by adjusting `prospector.scanner.fingerprint.length` (minimum value is 64). Keep in mind, the smaller the range, the more likely it is to hit a collision, so the fingerprint (ID) is not unique anymore. If this happens only the first file with a collided ID will be ingested.
+
+This file identity covers most of the use-cases. It's OS-agnostic, supports any file system including network shares and it makes sure to avoid data duplication or data loss.
```

Comment:
I'd make it all one sentence:
```suggestion
This file identity covers most of the use-cases, it's OS-agnostic, supports any file system including network shares and it makes sure to avoid data duplication or data loss.
```